apache/superset
# chore(config): Migrating `ENABLE_JAVASCRIPT_CONTROLS` from app config to a feature flag

### SUMMARY

This PR moves the boolean enablement of the JS controls (i.e. custom tooltips in the DeckGL viz) from app configuration to feature flags. Two main reasons: 
1) To better support deployments who manage feature flags on the fly... this seems more fitting in that area of configuration
2) To move the feature (with its self-admitted risks, and unknown level of community adoption) to the Feature Flag lifecycle, wherein we may more systematically either secure the feature's development and enablement, or deprecate it and replace it with a newer feature (e.g. jinja/markdown tooltips).

This is considered a breaking change since certain deployments may rely on enabling/disabling this feature, and they may have to adapt their configuration to set it in a new place.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
n/a

### TESTING INSTRUCTIONS
Turn on the flag, and see that it works! We can (and shall!) do this on an ephemeral environment on this PR.

### ADDITIONAL INFORMATION


- [x] Has associated issue: Fixes #18909 
- [ x ] Removes existing feature or API _well, kind of_
